server/channels/store/sqlstore/post_store.go

```diff
@@ -2706,6 +2758,27 @@ func (s *SqlPostStore) GetRepliesForExport(rootId string) ([]*model.ReplyForExpo
 		return nil, errors.Wrap(err, "failed to find Posts")
 	}
 
+	postIds := make([]string, len(posts))
+	for i, post := range posts {
+		postIds[i] = post.Id
+	}
+
+	flaggedMap, err := s.getFlaggedPostUsernames(postIds)
```

Comment:
I think I see the reason now. This same method is being called from multiple places. So instead of modifying multiple SQL queries, you just created a new one and called it from all these places.

I think in this case, it might be worth trading off the code maintenability for some performance boost. Mainly because we are joining with the posts table.